src/Transaction.proto

```diff
@@ -36,16 +41,22 @@ package Catalyst.Protocol.Transaction;
  * CFEntries: 0 field for non-confidential transaction
  * Signature: 64 bytes
  * EntryRangeProof: 0 field for non-confidential transaction
+ * Init: 0 field for smart contract deployment
+ * Data: 0 field for smart contract calls
+ * From: PubKey: 32 bytes, (255 bits or 256 with last bit to 0) public key (account address derived from the public key)
  */
 message TransactionBroadcast {
-    uint32 Version = 1;
+	TransactionType TransactionType = 1;
     google.protobuf.Timestamp TimeStamp = 2;
     uint64 TransactionFees = 3;
     uint64 LockTime = 4;
     repeated STTransactionEntry STEntries = 5;
```

Comment:
So now that we aren't doing aggregated signatures, we would normally only have one STTransactionEntry right, unless the sender is creating more than one transaction? Or would we normally have two, one for each account being updated by the transaction?

Reply:
There would be only 1 sig as there is 1 from address. So that sig will be generated by the "from" private key. The STEntries are entries to where the transaction is go to.

Reply:
Okay cool. This all still works with the confidential transactions as far as I can tell.